pkg/registry/core/pod/strategy.go

```diff
@@ -205,6 +211,52 @@ func (podStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Ob
 	return validation.ValidatePodStatusUpdate(obj.(*api.Pod), old.(*api.Pod))
 }
 
+type podEphemeralContainersStrategy struct {
+	podStrategy
+}
+
+var EphemeralContainersStrategy = podEphemeralContainersStrategy{Strategy}
+
+func mergeEphemeralContainers(oldContainers, newContainers []api.EphemeralContainer) []api.EphemeralContainer {
+	if len(newContainers) > len(oldContainers) {
+		mergedContainers := make([]api.EphemeralContainer, len(oldContainers), len(newContainers))
+		copy(mergedContainers, oldContainers)
+		for _, ec := range newContainers[len(oldContainers):] {
+			ec.Status = nil // Ephemeral Container status cannot be set using this interface
+			mergedContainers = append(mergedContainers, ec)
+		}
+		return mergedContainers
+	}
+	return oldContainers
+}
+
+// This strategy allows only a very specific operation: append to EphemeralContainers. Everything else is discarded.
```

Comment:
We should make this (i.e., append-only) clear in the the user-facing documentation since the update behavior is very different from regular containers, and there wouldn't be any validation error.

Reply:
Good point. I've added a note to this effect in the PodStatus docs.

That the append-only semantics are enforced silently was an arbitrary choice to match how updates by users to `PodStatus` are silently dropped (`EphemeralContainer` also contains a status).  It may be more user friendly here to instead fail validation if the append-only constraint is violated.